lib/problems.rb

```diff
@@ -1,13 +1,37 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  stack = Stack.new 
+  twosomes = {
+    "[" => "]",
+    "(" => ")",
+    "{" => "}"
+  }
+
+  string.each_char do |char|
+    if twosomes.values.include?(char)
```

Comment:
Note about `.include?` is an O(n) operation and unnecessary in a hash.

 ```suggestion
    if twosomes[char]
```